core/src/main/scala/frameless/CatalystBitwise.scala

```diff
@@ -0,0 +1,44 @@
+package frameless
+
+import scala.annotation.implicitNotFound
+
+/** Types that can be bitwise ORed, ANDed, or XORed by Catalyst.
+  *
+  * providing instances with implementations for bitwise operations since in the tests
```

Comment:
I think we should only have these implementations in tests, for instance by having "companion" test type classes. Have a look at what's done in [this test](https://github.com/typelevel/frameless/blob/master/dataset/src/test/scala/frameless/functions/AggregateFunctionsTests.scala#L95-L123) for an example.

Reply:
Sounds good. I'll refactor in the next 24 hours. Any other suggestions?

Reply:
Everything looks fine otherwise, you can maybe squash it into a single commit once you're done :)